### ISSUE TITLE
Check WCAG 2.2 conformity on `sd-navigation-item` represent a link AND parent item with children

### ISSUE THREAD

**Issue by karlbaumhauer** [solid-design-system/solid#844]
## Description
The `sd-navigation-item` can EITHER be a **link** or **have children** to prevent having 2 responsibilities/behaviors at the same time. This was decided when defining the component due to bad UX and a11y concerns. 

In the [JIRA Ticket #24](https://jira.apps.union-investment.de/browse/PSOLIDDS-24), the CMS team requested this feature to be implemented. 

## Task
In order to provide fully accessible components, we need to investigate if this would even be possible and still pass WCAG 2.2 standard requirements. 
Already defined challenges:
- [x] usability on mobile
- [x] screenreader usability
- [ ] clickable area sizes with long navigation item texts

## Acceptance Criteria:

**Icon Click Area:**
- [x] The clickable area for the icon should only include the icon itself and its touch area.

**Label Handling for Links:**
- [x] If a link is provided, the label must be replaced with sd-link. Necessary style adjustments should be made to ensure visual consistency.

**Focus State:**
- [x]  Implement a focus state for both the link and the icon to clearly indicate their different functionalities when both are present.

**Accessibility Testing:**
- [ ] Conduct comprehensive accessibility tests to ensure the component meets all accessibility standards and provides a good user experience.

- [ ] inform @karlbaumhauer to close the respective JIRA ticket to inform stakeholder

## Subtasks
- [ ] #1103 
- [x] https://github.com/solid-design-system/figma/issues/180

## DoR
- [x] Item has business value
- [x] All subtasks have been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] All subtasks have been closed

**Comment by yoezlem** [comment]
@coraliefeil since I was just asked again about the status of the ticket, I would like to ask you, for transparency reasons, to provide an update after your discussion with Damaris.
Thank you!

**Comment by coraliefeil** [comment]
Dev issue is ready for code.

**Comment by yoezlem** [comment]
@auroraVasconcelos if you're done with one of the AC, please check it! 🙂
